### DISCUSSION TITLE
UX ideas

### DISCUSSION THREAD

**Discussion by tarmotubro** [origin-energy/java-snapshot-testing#102] [Ideas]
Hey! Few ideas to improve usability (stolen from jest snapshot testing):

1. Should not require snapshot.properties to run (can use default settings)
2. After snapshot compare fails, could offer clickable links in IDE log window like 'Compare diff' (opens IDE compare view), 'replace old snapshot with new'

**Comment by jackmatt2** [comment]
1. I did consider this when implementing this feature - ultimately I decided against it for the following reasons.
  - Different languages Kotlin/Groovy use different src directories src/test/kotlin & src/test/groovy - I wanted to make it crystal clear what was being used without any sort of magic going on.
  - Other features such as CI detection, serializers, comparitors & reporters.  I wanted to keep explicit so new users know exactly what is being used.  If I decided to change the default in a newer version - their tests would start breaking and they would have no idea why.  By keeping it explicit I am free to change the defaults without breaking existing clients.
  
In the end it was a weigh up between being explicit and convenience.  I went with being explicit as it makes their lives easier and gives me the ability to make changes for the very small cost of adding a single JSON file.

2. I think there is a java library that will do this and automatically integrate with the IDE - I'll need to check though.  In this case it's changing the underlying reported to a different implementation which would not be a breaking change.
